### PR TITLE
Add an e2e test job for cloud-provider-azure

### PR DESCRIPTION
**Pull request by karataliu**
Setup for an e2e job. 

If this looks good, @BenTheElder could you please suggest where to supply the preset secret? I can provide the content.

### CODE REVIEW

File: jobs/config.json [hunk 1]
```diff
@@ -12306,6 +12306,18 @@
       "sig-apps"
     ]
   },
+  "pull-cloud-provider-azure-e2e": {
+    "args": [
+      "make",
```
Comment (BenTheElder) [inline jobs/config.json]: I think this will do as a start, but we might want to bring it a bit more in line with how the other e2e tests are run in CI.

**Comment by BenTheElder** [comment]
> If this looks good, @BenTheElder could you please suggest where to supply the preset secret? I can provide the content.

Sure, just privately contact me, or @krzyzacy, @cjwagner and we'll add it to the build cluster(s). 
We all have slack handles matching github and our emails are respectively [google.com]: bentheelder@, senlu@, cole@.

File: prow/config.yaml [hunk 2]
```diff
@@ -610,6 +637,52 @@ presubmits:
...
+        - "--root=/go/src"
+        - "--timeout=100"
+        - "--upload=gs://kubernetes-jenkins/pr-logs"
+        env:
+        - name: DOCKER_IN_DOCKER_ENABLED
+          value: 'true'
+        securityContext:
+          privileged: true
+          capabilities:
+            add: ["SYS_ADMIN"]
```
Comment (BenTheElder) [inline prow/config.yaml]: I don't think you need SYS_ADMIN here...? The other dind jobs have done fine with just `privileged`.

**Comment by cjwagner** [comment]
> cole@

colew@google.com
@cjwagner elsewhere

**Comment by BenTheElder** [comment]
oops, dropped that w :-)

**Comment by karataliu** [comment]
@BenTheElder Yeah, you're right, some of the additions are not required. I've updated the PR, PTAL.
For presets, will contact you later by email.

**Pull request merged**